extensions/Worker.Extensions.EventGrid/src/TypeConverters/EventGridCloudEventConverter.cs

```diff
@@ -2,32 +2,27 @@
 // Licensed under the MIT License. See License.txt in the project root for license information.
 
 using System;
-using System.Threading.Tasks;
+using System.Text.Json;
 using Azure.Messaging;
 using Microsoft.Azure.Functions.Worker.Converters;
 
 namespace Microsoft.Azure.Functions.Worker.Extensions.EventGrid.TypeConverters
 {
     /// <summary>
-    /// Converter to bind to CloudEvent or CloudEvent[] parameter.
+    /// Converter to bind to <see cref="CloudEvent" /> or <see cref="CloudEvent[]" /> type parameters.
     /// </summary>
     [SupportedConverterType(typeof(CloudEvent))]
     [SupportedConverterType(typeof(CloudEvent[]))]
-    internal class EventGridCloudEventConverter: IInputConverter
+    internal class EventGridCloudEventConverter: EventGridConverterBase
     {
-        public ValueTask<ConversionResult> ConvertAsync(ConverterContext context)
+        protected override ConversionResult ConvertCore(Type targetType, string json)
         {
-            if (context is null)
+            if (targetType != typeof(CloudEvent) && targetType != typeof(CloudEvent[]))
```

Comment:
Do we not support `IEnumerable<CloudEvent>`?

Reply:
We do not; this would be a nice to have post-parity deadline (I think we have an issue already tracking these improvements)